codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/presignable/PresignableUrlIntegration.kt

```diff
@@ -150,7 +144,14 @@ class PresignableUrlIntegration(private val presignedOperations: Map<String, Set
         operationMiddlewareCopy.removeMiddleware(op, MiddlewareStep.SERIALIZESTEP, "OperationInputHeadersMiddleware")
         operationMiddlewareCopy.removeMiddleware(op, MiddlewareStep.FINALIZESTEP, "ContentLengthMiddleware")
         operationMiddlewareCopy.removeMiddleware(op, MiddlewareStep.FINALIZESTEP, "AWSSigningMiddleware")
-        operationMiddlewareCopy.appendMiddleware(op, AWSSigningMiddleware(::customSigningParameters, context.model, context.symbolProvider))
+        val opID = op.id.toString()
+        val params = AWSSigningParams(
+            useSignatureTypeQueryString = true,
+            forceUnsignedBody = opID == "com.amazonaws.s3#PutObject" || opID == "com.amazonaws.s3#GetObject",
```

Comment:
A comment describing why this logic is needed here would be super helpful (since it's not obvious why its here and not also included in the other AWSSigningParams instances), especially for our future selves.

Reply:
trivial, create a set of operation ids than plain || check.

Reply:
Does this change only affect these two ops or affect other operations also?

Reply:
I refactored this check into `listof().contains()` and added a comment that this logic should be overhauled when new resigned URL types are added because while this logic is manageable for three URL types, it will get ugly fast if more are added.